### ISSUE TITLE
topology tests don't run in debug mode

### ISSUE THREAD

**Issue by avikivity** [scylladb/scylladb#16225]
It was reported that topology tests don't run in debug mode. This means that they're susceptible to use-after-free and other undefined behavior.

**Comment by avikivity** [comment]
A possible way to speed up topology under debug is to reduce num_tokens to 8.

**Comment by nyh** [comment]
> A possible way to speed up topology under debug is to reduce num_tokens to 8.

Note these tests already use num_tokens = 16 (see test/pylib/scylla_cluster.py), so I don't know how impressive the further reduction to 8 will be.

**Comment by kbr-scylla** [comment]
> It was reported that topology tests don't run in debug mode. 

Most of them do.

**Comment by kbr-scylla** [comment]
topology_custom/suite.yaml:
```
skip_in_debug:
  - test_shutdown_hang
  - test_replace
  - test_old_ip_notification_repro
  - test_node_shutdown_waits_for_pending_requests
```

+

topology_experimental_raft/suite.yaml:
```
skip_in_debug:
  - test_cdc_generation_clearing
  - test_cdc_generation_publishing
```

+

```python3
@skip_mode('debug', 'aarch64/debug is unpredictably slow', platform_key='aarch64')
async def test_cannot_add_new_node(manager: ManagerClient, raft_op_timeout: int) -> None:

...

@skip_mode('debug', 'aarch64/debug is unpredictably slow', platform_key='aarch64')
async def test_quorum_lost_during_node_join(manager: ManagerClient, raft_op_timeout: int) -> None:

...

@skip_mode('debug', 'aarch64/debug is unpredictably slow', platform_key='aarch64')
async def test_quorum_lost_during_node_join_response_handler(manager: ManagerClient, raft_op_timeout: int) -> None:

...

@skip_mode('debug', 'aarch64/debug is unpredictably slow', platform_key='aarch64')
async def test_cannot_run_operations(manager: ManagerClient, raft_op_timeout: int) -> None:
```

+

```python3
@skip_mode('debug', 'test requires nodes to be started quickly')
async def test_writes_to_old_previous_cdc_generation(request, manager: ManagerClient):
```

that is the entire list

**Comment by kbr-scylla** [comment]
- `test_shutdown_hang` is a regression test to reproduce very specific issue -- it reproduced outside debug, so no point running it in debug
- same for `test_old_ip_notification_repro`
- same for `test_node_shutdown_waits_for_pending_requests`

We should consider re-enabling `test_replace` in debug. (Perhaps at the same time, split the file with its test cases into multiple files.) Let's see what the slow-down will be.

The other listed tests are timing dependent. Some of them are only disabled for ARM debug for its extreme slowness.

So in summary only `test_replace` is up for consideration

**Comment by kbr-scylla** [comment]
Let's also use this opportunity to migrate from `skip_in_*` to `@skip_mode`
